splunk/common-files/Dockerfile

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018 Splunk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG SPLUNK_BASE_IMAGE=base-debian-10
+
+#
+# Download and unpack Splunk Enterprise
+#
+FROM ${SPLUNK_BASE_IMAGE}:latest as package
+ARG SPLUNK_BUILD_URL
+COPY splunk/common-files/splunk-minimal-exclude.list /tmp
+RUN echo "Downloading Splunk and validating the checksum at: ${SPLUNK_BUILD_URL}"
+RUN wget -qO /tmp/splunk.tgz ${SPLUNK_BUILD_URL}
+RUN wget -qO /tmp/splunk.tgz.md5 ${SPLUNK_BUILD_URL}.md5
+RUN test $(md5sum /tmp/splunk.tgz | sed 's,\([a-z0-9]*\).*,\1,') = $(cat /tmp/splunk.tgz.md5 | sed 's,MD5.*=.\([a-z0-9]*\).*,\1,')
+RUN mkdir -p /opt/splunk/var /minimal/splunk/var /extras/splunk/var
+RUN tar -C /minimal/splunk --strip 1 --exclude-from=/tmp/splunk-minimal-exclude.list -zxf /tmp/splunk.tgz
+RUN tar -C /extras/splunk --strip 1 --wildcards --files-from=/tmp/splunk-minimal-exclude.list -zxf /tmp/splunk.tgz
+RUN mv /minimal/splunk/etc /minimal/splunk-etc
+RUN mv /extras/splunk/etc /extras/splunk-etc
+RUN mkdir -p /minimal/splunk/etc/apps/_splunk_config /extras/splunk/etc/apps/_splunk_config
+COPY splunk/common-files/apps /extras/splunk-etc/apps/
+
+
+#
+# Minimal Splunk base image with many files excluded, intended for internal and experimental use
+#
+FROM ${SPLUNK_BASE_IMAGE}:latest as minimal
+LABEL maintainer="support@splunk.com"
+ENV SPLUNK_HOME=/opt/splunk \
+    SPLUNK_GROUP=splunk \
+    SPLUNK_USER=splunk \
+    TMPSPLUNKDIR=${SPLUNK_HOME}/tmp \
+    TMPETCDIR=${TMPSPLUNKDIR}/etc
```

Comment:
i think these need to be separated into separate ENV statements. TMPSPLUNKDIR is set to `/tmp` and TMPETCDIR is set to `/etc`